analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SillyBitwiseOperation.CSharp9.cs

```diff
@@ -24,11 +27,11 @@
 
 nuint bitMaskNuint = 0x010F;
 nuint resultNuint;
-nuint nuOne = 1;
-nuint nuZero = 0;
+const nuint nuZero = 0;
 
-resultNuint = bitMaskNuint & + + +nuOne;    // FN
+resultNuint = bitMaskNuint | + + +nuZero;   // Noncompliant
 resultNuint = bitMaskNuint & nuZero;        // Compliant
+resultNuint = bitMaskNuint ^ 0;             // Noncompliant
```

Comment:
you can also add tests with `UIntPtr.Zero` and `IntPtr.Zero;`

Reply:
I'm not sure I understand this. Operators `&`, `|` and `^` are not defined for `UIntPtr` and `IntPtr`. Do I miss something?

Reply:
This builds on my machine:
```
        nuint x = 1;
        x = x ^ UIntPtr.Zero;
```

Reply:
Ah, my bad, I've applied `^` operator to a mix of nint and `UIntPtr`.